posthog/models/feature_flag.py

```diff
@@ -227,14 +254,14 @@ def is_condition_match(self, feature_flag: FeatureFlag, condition: Dict, conditi
                 condition_match = self._condition_matches(feature_flag, condition_index)
 
             if not condition_match:
-                return False
+                return False, FeatureFlagMatchReason.NO_CONDITION_MATCH
             elif rollout_percentage is None:
-                return True
+                return True, FeatureFlagMatchReason.CONDITION_MATCH
 
         if rollout_percentage is not None and self.get_hash(feature_flag) > (rollout_percentage / 100):
-            return False
+            return False, FeatureFlagMatchReason.OUT_OF_ROLLOUT_BOUND
 
-        return True
+        return True, FeatureFlagMatchReason.CONDITION_MATCH
```

Comment:
Is this redundant now? Do we need both bool and reason

Reply:
Would this return an empty reason otherwise? I think it's nice to have both 😃

Reply:
Was just easier this way to keep things explicit. Another benefit is that it changes the method signature, so everything that is supposed to fail, fails loudly right now, vs. finding out at runtime.